PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskSubJetFraction.cxx

```diff
@@ -1947,6 +1960,48 @@ std::vector<fastjet::PseudoJet> AliAnalysisTaskSubJetFraction::AddExtraProng(std
   fInputVectors.push_back(ExtraTrack); 
   return fInputVectors;
 }
+
+std::vector<fastjet::PseudoJet> AliAnalysisTaskSubJetFraction::AddkTTracks(AliEmcalJet *Jet,std::vector<fastjet::PseudoJet> fInputVectors, Double_t QHat,Double_t XLength, Int_t NAdditionalTracks)
+{
+//here add N tracks with random phi and eta and theta according to bdmps distrib.
+  fastjet::PseudoJet Lab_Jet;
+  Lab_Jet.reset(Jet->Px(),Jet->Py(),Jet->Pz(),Jet->E());
+  // Double_t Omega_C=0.5*QHat*XLength*XLength/0.2;
+  // Double_t Theta_C=TMath::Sqrt(12*0.2/(QHat*TMath::Power(XLength,3)));
+  Double_t xQs=TMath::Sqrt(QHat*XLength); //1,2				
+    //cout<<"medium parameters "<<Omega_C<<" "<<Theta_C<<" "<<xQs<<endl;
+
+   for(Int_t i=0;i<NAdditionalTracks;i++){
+
+     Double_t kT_Scale,Limit_Min,Limit_Max;
+    
+    //generation of kT according to 1/kT^4, with minimum QS=2 GeV and maximum ~sqrt(ptjet*T)	
+     kT_Func= new TF1("kT_Func","1/(x*x*x*x)",xQs,10000);
```

Comment:
Where do you delete this function? Rather make local variable 

TF1 kT_Func(...);